### PR TITLE
Make error handling in move shard tests more robust.

### PR DESCRIPTION
**Pull request by neunhoef**

### CODE REVIEW

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/2183/

File: tests/js/server/resilience/moving-shards-cluster.js [hunk 1]
```diff
@@ -119,11 +119,21 @@ function MovingShardsSuite () {
...
+    if (res.statusCode !== 200) {
+      return {cleanedServers:[]};
     }
     var body = res.body;
     if (typeof body === "string") {
-      body = JSON.parse(body);
+      try {
+        body = JSON.parse(body);
+      } catch (err2) {
+      }
```
Comment (hkernbach) [inline tests/js/server/resilience/moving-shards-cluster.js]: err2 is ignored and never handled. Is that expected behaviour?

File: tests/js/server/resilience/moving-shards-with-arangosearch-view-cluster-grey.js [hunk 2]
```diff
@@ -174,11 +174,21 @@ function MovingShardsWithViewSuite (options) {
...
+    if (res.statusCode !== 200) {
+      return {cleanedServers:[]};
     }
     var body = res.body;
     if (typeof body === "string") {
-      body = JSON.parse(body);
+      try {
+        body = JSON.parse(body);
+      } catch (err2) {
+      }
```
Comment (hkernbach) [inline tests/js/server/resilience/moving-shards-with-arangosearch-view-cluster-grey.js]: Same as above: err2 is ignored and never handled. Is that expected behaviour?

File: tests/js/server/resilience/moving-shards-cluster.js (hunk 1 above)
Comment (neunhoef) [reply tests/js/server/resilience/moving-shards-cluster.js]: Yes, if JSON.parse throws, then body is not changed and this is checked further down.

Comment (neunhoef) [reply tests/js/server/resilience/moving-shards-cluster.js]: Maybe this is not the most elegant way, but it would do the job.

Comment (hkernbach) [reply tests/js/server/resilience/moving-shards-cluster.js]: Ok, then LGTM

**Pull request merged**